### PR TITLE
API docs for Libplanet.Action namespace

### PR DESCRIPTION
**Pull request by dahlia**
Wrote API docs for `Libplanet.Action` namespace.  It also made the return type of `IAction.RequestStates()` method immutable.

There's some example code as well in docs of `Libplanet.Action.IAction` interface.  Please review carefully that code as well.

Preview: <https://planetarium.github.io/libplanet.net/pulls/63/api/Libplanet.Action.html>

### CODE REVIEW

**Comment by dahlia** [comment]
@longfin I rebased it upon the current master.  Please approve this again.

File: Libplanet/Action/IAction.cs [hunk 1]
```diff
@@ -1,16 +1,242 @@
...
+        /// </summary>
+        /// <param name="context">A context object which holds addresses that
+        /// signed and receives the transaction this action belongs to, states
+        /// immediately before this action is executed, and a PRNG object
+        /// which procudes deterministic random numbers.
+        /// See <see cref="IActionContext"/> for details.</param>
+        /// <returns>A map of <see cref="Address"/>es to update their states.
+        /// Only addresses this map contains are updated in the global states.
+        /// That means, addresses this map misses are unchanged.</returns>
+        /// <remarks>This method should be replayable (deterministic).
```
Comment (kijun) [inline Libplanet/Action/IAction.cs]: remove this line?

File: Libplanet/Action/IAction.cs [hunk 2]
```diff
@@ -1,16 +1,242 @@
...
+        /// object in itself, as its properties (or fields).</para>
+        /// <para>A returned <see cref="AddressStateMap"/> object functions as
+        /// a delta which shifts from previous states to next states.</para>
+        /// </summary>
+        /// <param name="context">A context object which holds addresses that
+        /// signed and receives the transaction this action belongs to, states
+        /// immediately before this action is executed, and a PRNG object
+        /// which procudes deterministic random numbers.
+        /// See <see cref="IActionContext"/> for details.</param>
+        /// <returns>A map of <see cref="Address"/>es to update their states.
```
Comment (kijun) [inline Libplanet/Action/IAction.cs]: A map of addresses with changed states. Only those addresses returned by this map will have their states updated globally.

File: Libplanet/Action/IAction.cs [hunk 3]
```diff
@@ -1,16 +1,242 @@
...
+    /// result.
+    /// <para>A &#x201c;class&#x201d; which implements this interface is
+    /// analogous to a function, and its instance is analogous to a
+    /// <a href="https://en.wikipedia.org/wiki/Partial_application">partial
+    /// function application</a>, in other words, a function with some bound
+    /// arguments.  Those parameters that will be bound at runtime should be
+    /// represented as fields or properties in an action class, and bound
+    /// argument values to these parameters should be received through
+    /// a constructor parameters of that class.</para>
+    /// <para>By convention, action classes are named verbally, e.g.,
```
[the remark this answers is not in the document]
Comment (dahlia) [reply Libplanet/Action/IAction.cs]: @kijun Can I omit “are” here?

File: Libplanet/Action/ActionTypeAttribute.cs [hunk 4]
```diff
@@ -4,21 +4,46 @@
 
 namespace Libplanet.Action
 {
+    /// <summary>
+    /// Indicates that an action class (i.e., a class implementing
+    /// <see cref="IAction"/>) can be wired into transactions and blocks.
```
[the remark this answers is not in the document]
Comment (dahlia) [reply Libplanet/Action/ActionTypeAttribute.cs]: I meant, it indicates that an action class can be held by transactions and blocks.  I'll fix it that way.

**Comment by dahlia** [comment]
Reflected corrections.  @longfin @kijun Please approve this again.

**Pull request merged**